### ISSUE TITLE
View of MV slower than querying directly?

### ISSUE THREAD

**Issue by deklanw** [ClickHouse/ClickHouse#11937]
```sql

CREATE MATERIALIZED VIEW player_champ_counts_mv
ENGINE = AggregatingMergeTree() ORDER BY (patch_num, my_account_id, champ_id)
as select patch_num, my_account_id, champ_id, countState() as c_state
from full_info
group by patch_num, my_account_id, champ_id;

create view player_champ_counts as
select patch_num, my_account_id, champ_id, countMerge(c_state) as c
from player_champ_counts_mv
group by patch_num, my_account_id, champ_id;
```

2.4s:
`select * from player_champ_counts`

900ms:
```
select patch_num, my_account_id, champ_id, countMerge(c_state) as c
from player_champ_counts_mv
group by patch_num, my_account_id, champ_id;
```

In other words, just inlining the definition of the view is doing something different than querying the view itself.

**Comment by den-crane** [comment]
It needs clarification and reproduction.

1. CH version.
2. show create table full_info
3. data sample for full_info

**Comment by deklanw** [comment]
@den-crane 

`ClickHouse server version 20.4.5.36 (official build).`

```sql
CREATE TABLE full_info (
	`my_game_id` UInt64,
	`game_id` UInt64,
	`region` String,
	`game_tier` UInt8,
	`patch_full` String,
	`patch_num` UInt16,
	`game_start_time` DateTime64(3),
	`my_account_id` UInt64,
	`champ_id` UInt8,
	`canonical_role` UInt8,
	`won` UInt8,
	`highest_rank_in_game` UInt8
)
ENGINE = MergeTree()
ORDER BY tuple()
```

`SELECT * FROM full_info LIMIT 20;`
```
my_game_id|game_id   |region|game_tier|patch_full    |patch_num|game_start_time        |my_account_id|champ_id|canonical_role|won|highest_rank_in_game|
----------|----------|------|---------|--------------|---------|-----------------------|-------------|--------|--------------|---|--------------------|
       114|1978239875|BR1   |       60|10.12.325.3360|     1012|2014-08-06   :09:29.000|       703916|      82|             1|  0|                   0|
       114|1978239875|BR1   |       60|10.12.325.3360|     1012|2014-08-06   :09:29.000|       663109|     238|             3|  0|                   0|
       114|1978239875|BR1   |       60|10.12.325.3360|     1012|2014-08-06   :09:29.000|       617173|     145|             4|  0|                   0|
       114|1978239875|BR1   |       60|10.12.325.3360|     1012|2014-08-06   :09:29.000|       702115|     107|             5|  0|                   1|
       114|1978239875|BR1   |       60|10.12.325.3360|     1012|2014-08-06   :09:29.000|       918117|      60|             2|  0|                   0|
       114|1978239875|BR1   |       60|10.12.325.3360|     1012|2014-08-06   :09:29.000|       765669|      58|             1|  1|                   1|
       114|1978239875|BR1   |       60|10.12.325.3360|     1012|2014-08-06   :09:29.000|       672571|     236|             3|  1|                   1|
       114|1978239875|BR1   |       60|10.12.325.3360|     1012|2014-08-06   :09:29.000|       595553|     173|             4|  1|                   0|
       114|1978239875|BR1   |       60|10.12.325.3360|     1012|2014-08-06   :09:29.000|       919079|     241|             5|  1|                   1|
       114|1978239875|BR1   |       60|10.12.325.3360|     1012|2014-08-06   :09:29.000|      1136555|      20|             2|  1|                   0|
       290| 235855467|JP1   |       50|10.12.325.3360|     1012|2014-06-06 p�:51:12.000|       355828|     145|             4|  1|                   0|
       290| 235855467|JP1   |       50|10.12.325.3360|     1012|2014-06-06 p�:51:12.000|      1579000|       8|             1|  1|                   0|
       290| 235855467|JP1   |       50|10.12.325.3360|     1012|2014-06-06 p�:51:12.000|      1563001|      81|             3|  1|                   0|
       290| 235855467|JP1   |       50|10.12.325.3360|     1012|2014-06-06 p�:51:12.000|       563780|      48|             2|  1|                   0|
       290| 235855467|JP1   |       50|10.12.325.3360|     1012|2014-06-06 p�:51:12.000|        38667|      26|             5|  1|                   1|
       290| 235855467|JP1   |       50|10.12.325.3360|     1012|2014-06-06 p�:51:12.000|       252279|     122|             1|  0|                   0|
       290| 235855467|JP1   |       50|10.12.325.3360|     1012|2014-06-06 p�:51:12.000|       356250|      90|             3|  0|                   0|
       290| 235855467|JP1   |       50|10.12.325.3360|     1012|2014-06-06 p�:51:12.000|       102144|     111|             5|  0|                   0|
       290| 235855467|JP1   |       50|10.12.325.3360|     1012|2014-06-06 p�:51:12.000|       102231|      11|             4|  0|                   0|
       290| 235855467|JP1   |       50|10.12.325.3360|     1012|2014-06-06 p�:51:12.000|       395535|     121|             2|  0|                   0|
```

```
name                  |value      |
----------------------|-----------|
extremes              |0          |
use_uncompressed_cache|0          |
load_balancing        |random     |
max_result_rows       |200        |
result_overflow_mode  |break      |
max_memory_usage      |10000000000|
```

**Comment by den-crane** [comment]
WA: set experimental_use_processors=0


20.3.12.112, 20.4.5.36, 20.4.6.53 affected. 
20.6.1.3853 is not affected.

repro:


```

CREATE TABLE dw.player_champ_counts
(   patch_num UInt64,
    c_state AggregateFunction(count))
ENGINE = AggregatingMergeTree()
ORDER BY (patch_num);

insert into player_champ_counts
select number, countState() from numbers(10000000) group by number;

create view player_champ_counts_v as
select patch_num, countMerge(c_state) as c
from player_champ_counts
group by patch_num;

select patch_num, countMerge(c_state) as c from player_champ_counts group by patch_num format Null;
0 rows in set. Elapsed: 0.447 sec. Processed 10.00 million rows, 303.59 MB (22.36 million rows/s., 678.71 MB/s.)

set experimental_use_processors=1
select * from player_champ_counts_v format Null;
0 rows in set. Elapsed: 1.399 sec. Processed 10.00 million rows, 303.59 MB (7.15 million rows/s., 216.98 MB/s.)
0 rows in set. Elapsed: 1.400 sec. Processed 10.00 million rows, 303.59 MB (7.14 million rows/s., 216.84 MB/s.)

set experimental_use_processors=0
select * from player_champ_counts_v format Null;
0 rows in set. Elapsed: 0.430 sec. Processed 10.00 million rows, 303.59 MB (23.23 million rows/s., 705.32 MB/s.)
0 rows in set. Elapsed: 0.396 sec. Processed 10.00 million rows, 303.59 MB (25.26 million rows/s., 766.84 MB/s.)
```

**Comment by deklanw** [comment]
Thanks for the quick follow-up.

I can confirm the issue is fixed with `20.6.1.3853`
